server/infinispan/src/test/java/org/hawkular/apm/server/infinispan/InfinispanPropertyCacheTest.java

```diff
@@ -0,0 +1,123 @@
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.server.infinispan;
+
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.hawkular.apm.server.api.services.CacheException;
+import org.hawkular.apm.server.api.services.Property;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * @author Pavol Loffay
+ */
+public class InfinispanPropertyCacheTest extends AbstractInfinispanTest {
+
+    @Test
+    public void testGetNonExisting() {
+        InfinispanPropertyCache cache = createCache();
+
+        Property property = cache.get(null, "id");
+        Assert.assertNull(property);
+    }
+
+    @Test
+    public void testGetOne() throws CacheException {
+        InfinispanPropertyCache cache = createCache();
+
+        Property<Double> doubleProperty = new Property<>("id", 15D);
+        Property<String> stringProperty = new Property<>("id2", "str");
+
+        cache.store(null, doubleProperty);
+        cache.store(null, stringProperty);
+
+        Property propertyFromDb = cache.get(null, "id");
+        Assert.assertEquals(doubleProperty.getValue(), propertyFromDb.getValue());
+
+        propertyFromDb = cache.get(null, "id2");
+        Assert.assertEquals(stringProperty.getValue(), propertyFromDb.getValue());
+    }
+
+    @Test
+    public void testSystemProp() {
+        InfinispanPropertyCache cache = createCache();
+
+        System.setProperty("strProp", "foo");
+        Assert.assertEquals("foo", cache.get(null, "strProp").getValue());
+        System.clearProperty("strProp");
+
+        System.setProperty("intProp", "2");
+        Assert.assertEquals("2", cache.get(null, "intProp").getValue());
+        System.clearProperty("intProp");
+    }
+
+    @Test
+    public void testObservable() throws CacheException {
+        InfinispanPropertyCache cache = createCache();
+
+        DummyObserver observer = new DummyObserver();
+
+        Property<Integer> property = new Property<>("id1", 2);
+        cache.store(null, property);
+        cache.addObserver(null, "id1", observer);
+
+        cache.store(null, new Property<>("id1", 33));
+        cache.store(null, new Property<>("id1", 34));
+        Assert.assertEquals(2, observer.called.intValue());
+
+        cache.store(null, new Property<>("id1", 34));
+        Assert.assertEquals(2, observer.called.intValue());
+    }
+
```

Comment:
Can you create a test with the observer being added before the initial property value is stored?